### PR TITLE
properly handle Ace editor focus w/RStudio Server

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/rstudio/issues/4622.

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java [hunk 1]
```diff
@@ -4165,6 +4174,7 @@ private static final ExternalJavaScriptLoader getLoader(StaticDataResource relea
    private long lastModifiedTime_;
    private String yankedText_ = null;
    
+   private static AceEditor LAST_FOCUSED_EDITOR = null;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java]: Doesn't all-caps normally imply a constant?

Comment (kevinushey) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java]: Oops, you're right. Should this still be `s_`-prefixed as we do in C++ sources?

Comment (gtritchie) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java]: Our coding standards say yes, but seems to be very rare in the Java sources (I found one example).

**Comment by jmcphers** [comment]
If I'm reading this correctly, then clicking in the console, and then attempting to run a "editor must be focused" command will now cause that command to execute (as the last focused editor is still active). Is that correct/intentional?

**Comment by kevinushey** [comment]
That's intentional. For the purposes of these commands, the console is considered "just another editor" as opposed to specifically being a Source window editor.

That distinction could probably be made more clearly in the code, and it might be better to refer to this as the 'active' editor rather than the 'focused' editor, since in theory an editor (including the console) could remain active while temporarily losing focus (ie: as described in the associated bug report)

**Pull request merged**